src/interfaces/wallet.cpp

```diff
@@ -335,7 +335,7 @@ class WalletImpl : public Wallet
         LOCK(m_wallet->cs_wallet);
         auto mi = m_wallet->mapWallet.find(txid);
         if (mi != m_wallet->mapWallet.end()) {
-            num_blocks = locked_chain->getHeight().get_value_or(-1);
+            num_blocks = m_wallet->GetLastBlockHeight();
```

Comment:
`m_last_block_processed_height` is already initialized at `-1`, so behavior change is in case wallet isn't at tip right ? If yes, it should be more accurate.